graphql_compiler/tests/test_data_tools/data_tool.py

```diff
@@ -206,17 +225,29 @@ def generate_sql_integration_data(sql_test_backends):
             from_classname = uuid_to_class_name[edge_value['from_uuid']]
             edge_field_name = 'out_{}'.format(edge_name)
             join_descriptor = sql_schema_info.join_descriptors[from_classname][edge_field_name]
-            if join_descriptor.to_column != 'uuid':
-                raise NotImplementedError(u'Expected to_column to be uuid, found {} for {}'
-                                          .format(join_descriptor.to_column, edge_name))
-
-            existing_foreign_key_values = uuid_to_foreign_key_values.setdefault(
-                edge_value['from_uuid'], {})
-            if join_descriptor.from_column in existing_foreign_key_values:
-                raise NotImplementedError(u'The SQL backend does not support many-to-many edges.'
-                                          u'Found multiple edges of class {} from vertex {}'
-                                          .format(edge_name, edge_value['from_uuid']))
-            existing_foreign_key_values[join_descriptor.from_column] = edge_value['to_uuid']
+
+            is_from_uuid = join_descriptor.from_column == 'uuid'
+            is_to_uuid = join_descriptor.to_column == 'uuid'
+            if is_from_uuid == is_to_uuid:
+                raise NotImplementedError(u'Exactly one of the join columns was expected to'
+                                          u'be uuid. found {}'.format(join_descriptor))
+
+            if is_from_uuid:
+                existing_foreign_key_values = uuid_to_foreign_key_values.setdefault(
+                    edge_value['to_uuid'], {})
+                if join_descriptor.to_column in existing_foreign_key_values:
+                    raise NotImplementedError(u'The SQL backend does not support many-to-many edges.'
+                                              u'Found multiple edges of class {} from vertex {}'
+                                              .format(edge_name, edge_value['to_uuid']))
+                existing_foreign_key_values[join_descriptor.to_column] = edge_value['from_uuid']
+            elif is_to_uuid:
+                existing_foreign_key_values = uuid_to_foreign_key_values.setdefault(
+                    edge_value['from_uuid'], {})
+                if join_descriptor.from_column in existing_foreign_key_values:
+                    raise NotImplementedError(u'The SQL backend does not support many-to-many edges.'
+                                              u'Found multiple edges of class {} from vertex {}'
+                                              .format(edge_name, edge_value['from_uuid']))
+                existing_foreign_key_values[join_descriptor.from_column] = edge_value['to_uuid']
```

Comment:
Without this change, we get `NotImplementedError` when using the `Animal_ParentOf` edge, since it's `to_column` is not `uuid`.